codalab/worker/worker_run_state.py

```diff
@@ -276,22 +277,32 @@ def mount_dependency(dependency, shared_file_system):
         docker_image = run_state.resources.docker_image
         image_state = self.docker_image_manager.get(docker_image)
         if image_state.stage == DependencyStage.DOWNLOADING:
-            status_messages.append(
-                'Pulling docker image: ' + (image_state.message or docker_image or "")
-            )
+            msg = 'Pulling docker image %s' % docker_image
+            if isinstance(image_state.message, dict) and image_state.message['progressDetail']:
+                msg += '(%d%%)' % (
```

Comment:
I think it would be cleaner to instead just make sure that `image_state.message` is always a string (it's good practice in general not to have the same variable have multiple possible types in Python). This means that this code here would probably be moved over to `docker_image_manager.py` and you'd set `self._downloading[image_spec]['status']` equal to `"Pulling docker image (XX%)"`